src/PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Krivodonova.cpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Krivodonova.hpp"
+
+#include <cmath>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ScalarAdvection::Solutions {
+
+template <typename DataType>
+tuples::TaggedTuple<ScalarAdvection::Tags::U> Krivodonova::variables(
+    const tnsr::I<DataType, 1>& x, double t,
+    tmpl::list<ScalarAdvection::Tags::U> /*meta*/) const noexcept {
+  // map each grid points x(t) back to its initial position within [-1, 1] at
+  // t=0
+  auto x0 = make_with_value<tnsr::I<DataType, 1>>(x, 0.0);
+  get<0>(x0) = get<0>(x) - t;
+  for (size_t i = 0; i < get_size(get<0>(x)); ++i) {
+    auto& xi = get<0>(x0)[i];
+    xi = xi - 2.0 * floor(0.5 * (xi + 1.0));
+  }
+
+  // parameters for the initial profile (from the Krivodonova paper)
+  const double a{0.5};
+  const double z{-0.7};
+  const double delta{0.005};
+  const double alpha{10.0};
+  const double beta{log(2.0) / (36.0 * pow(delta, 2.0))};
```

Comment:
`pow` can be pretty slow (doesn't really matter in an analytic solution) but I'd suggest using `square`, which I think needs the include `Utilities/ConstantExpressions.hpp`